app/assets/javascripts/client/consensus/ConsensusService.js

```diff
@@ -0,0 +1,153 @@
+(function() {
+  'use strict';
+  angular.module('PDRClient')
+      .service('ConsensusService', ConsensusService);
+
+  ConsensusService.$inject = [
+    '$stateParams',
+    '$location',
+    '$q',
+    'Consensus',
+    'AnalysisConsensus',
+    'AnalysisResponse',
+    'Score',
+    'AnalysisConsensusScore',
+    'ProductEntry',
+    'DataEntry'
+  ];
+
+  function ConsensusService($stateParams, $location, $q, Consensus, AnalysisConsensus, AnalysisResponse, Score, AnalysisConsensusScore, ProductEntry, DataEntry) {
+    var service = this;
+
+    service.setContext = function(context) {
+      service.context = context;
+      return service;
+    };
+
+    service.extractId = function() {
```

Comment:
Neither: it's extracting the *relevant* ID. This `ConsensusService`, like others before it, is designed to abstract away the particularities of the entities it works with (in this particular case, assessments and analyses).
Note that there's only one of `$stateParams.assessment_id` or `$stateParams.analysis_id` at any given route, which is why this works (i.e. `$stateParams.assessment_id && $stateParams.analysis_id` is always `undefined`.